### PR TITLE
feat: coinbase wallet

### PR DESCRIPTION
**Pull request by asamere**
## Description

Adds support for Coinbase Wallet

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [X] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #

### Engineering

see functional testing from ops below

### Operations

**please note this is flagged on in all environments**

per @MBMaria below

- [x] Chrome incognito, with coinbase extension (on desktop) and Coinbase wallet app (on mobile).
- [x] Connected via firefox and brave on desktop using QR code.
- [x] Connected via safari on desktop using Coinbase extension
- [x] Connected with safari, firefox and brave browsers on mobile using coinbase wallet app

## Screenshots (if applicable)

image

image

### CODE REVIEW

File: src/context/WalletProvider/Coinbase/components/Connect.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+export interface CoinbaseSetupProps
+  extends RouteComponentProps<
+    {},
+    any, // history
+    LocationState
+  > {
+  dispatch: React.Dispatch<ActionTypes>
+}
+
+export const CoinbaseConnect = ({ history }: CoinbaseSetupProps) => {
```
Comment (0xdef1cafe) [inline src/context/WalletProvider/Coinbase/components/Connect.tsx:24]: we'll want to get history from a hook like this

https://github.com/shapeshift/web/blob/develop/src/components/Modals/NativeOnboarding/components/EasyToUse.tsx#L20

File: src/context/WalletProvider/Coinbase/components/Connect.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  const { dispatch, state, onProviderChange } = useWallet()
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState<string | null>(null)
+
+  // eslint-disable-next-line no-sequences
```
Comment (0xdef1cafe) [inline src/context/WalletProvider/Coinbase/components/Connect.tsx:29]: ser please there's a reason we have this lint rule

File: package.json [hunk 3]
```diff
@@ -92,15 +92,16 @@
...
-    "@shapeshiftoss/hdwallet-core": "1.50.3",
-    "@shapeshiftoss/hdwallet-keepkey": "1.50.3",
-    "@shapeshiftoss/hdwallet-keepkey-webusb": "1.50.3",
-    "@shapeshiftoss/hdwallet-keplr": "1.50.3",
-    "@shapeshiftoss/hdwallet-metamask": "1.50.3",
-    "@shapeshiftoss/hdwallet-native": "1.50.3",
-    "@shapeshiftoss/hdwallet-native-vault": "1.50.3",
-    "@shapeshiftoss/hdwallet-walletconnect": "1.50.3",
-    "@shapeshiftoss/hdwallet-xdefi": "1.50.3",
+    "@shapeshiftoss/hdwallet-coinbase": "1.50.4-alpha.1",
```
Comment (0xdef1cafe) [inline package.json:95]: do not merge until these are non-alpha versions

**Review by 0xdef1cafe — COMMENTED** [review]
looking grouse so far @asamere 

very small quality of life thing - closing the modal isn't handled gracefully, it just stays on `Pairing` with a spinner, we should probably handle this better.

File: src/context/WalletProvider/Coinbase/components/Connect.tsx (hunk 1 above)
Comment (0xdef1cafe) [reply src/context/WalletProvider/Coinbase/components/Connect.tsx:24]: as discussed this is a straight copypasta from metamask/keplr implementations of this era so i'm fine with this

**Comment by 0xdef1cafe** [comment]
converting to draft until https://github.com/shapeshift/hdwallet/pull/624/files# is merged and packages bumped

**Comment by MBMaria** [comment]
Tested connection on desktop as well as pairing from web on mobile. 

Chrome incognito, with coinbase extension (on desktop) and Coinbase wallet app (on mobile). 
Connected via firefox and brave on desktop using QR code. 
Connected via safari on desktop using Coinbase extension 
Connected with safari, firefox and brave browsers on mobile using coinbase wallet app

**Comment by MBMaria** [comment]
When connecting to coinbase via chrome incognito, it will only connect with the first wallet address . even though Coinbase wallet is set to the second wallet address, shapeshift still registers the first wallet. cant seem to get it to switch the second wallet. 
Using safari however will allow you to switch between wallets by selecting the second wallet in the extension then refreshing ss page.

**Comment by 0xdef1cafe** [comment]
> When connecting to coinbase via chrome incognito, it will only connect with the first wallet address . even though Coinbase wallet is set to the second wallet address, shapeshift still registers the first wallet. cant seem to get it to switch the second wallet. Using safari however will allow you to switch between wallets by selecting the second wallet in the extension then refreshing ss page.

as discussed here let's just ship it https://discord.com/channels/554694662431178782/1115490157769019423/1116169615261835264

File: src/context/WalletProvider/WalletProvider.tsx [hunk 4]
```diff
@@ -734,6 +738,14 @@ export const WalletProvider = ({ children }: { children: React.ReactNode }): JSX
                       1: getConfig().REACT_APP_ETHEREUM_NODE_URL,
                     },
                   }
+                case 'coinbase':
+                  return {
+                    appName: 'ShapeShift',
+                    appLogoUrl: 'https://avatars.githubusercontent.com/u/52928763?s=50&v=4',
+                    defaultJsonRpcUrl: getConfig().REACT_APP_ETHEREUM_NODE_URL,
+                    defaultChainId: 1,
+                    darkMode: false,
```
Comment (0xdef1cafe) [inline src/context/WalletProvider/WalletProvider.tsx:747]: @reallybeard can you plumb dark mode pref through here plz? 👯‍♂️

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**